### PR TITLE
Error out when two different lockfiles are present in the repo

### LINKED ISSUE #410: Error out if there are both npm and yarn lockfiles
npm5 is going to be released soon, which introduces a new lockfile-by-default just like yarn. This means that if you use yarn, and ever use npm to install something or vice-versa you'll end up with two different lockfiles.

To prevent subtle errors in prod this should result in a failed build on Heroku with a helpful error message.

### PR DESCRIPTION
**Pull request by jmorrell**
npm5 is going to be released soon, which introduces a new lockfile-by-default just like yarn. This means that if you use yarn, and ever use npm to install something or vice-versa you'll end up with two different lockfiles.

To prevent subtle errors in prod this should result in a failed build on Heroku with a helpful error message.

hyper 2017-05-25 22-01-05

Fixes #410

### CODE REVIEW

File: lib/failure.sh [hunk 1]
```diff
@@ -63,6 +63,32 @@ fail_dot_heroku_node() {
...
+
+       Both npm and yarn have created lock files for this application,
+       but only one can be used to install dependencies. Installing
+       dependencies using the wrong package manager can result in missing
+       packages or subtle bugs in production.
+
+       - To use npm to install your application's dependencies please delete
+         the yarn.lock file.
+
+         $ rm yarn.lock
```
Comment (hone) [inline lib/failure.sh]: should we say `git rm yarn.lock`?

File: lib/failure.sh [hunk 2]
```diff
@@ -63,6 +63,32 @@ fail_dot_heroku_node() {
...
+
+       - To use yarn to install your application's dependences please delete
+         the package-lock.json file.
+
+         $ rm package-lock.json
```
Comment (hone) [inline lib/failure.sh]: should we say `git rm package-lock.json`?

Comment (jmorrell) [inline lib/failure.sh]: +1 I had this at first, but wasn't sure so I changed it

Comment (hone) [inline lib/failure.sh]: I imagine most people are using git.

**Pull request merged**